### PR TITLE
[CA-884]Generate one more SA account and fix the test breakage

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java [hunk 1]
```diff
@@ -5,16 +5,29 @@
...
 public class IntegrationCredentials {
-  /** Path to the service account credentials file. */
-  private static final String GOOGLE_SERVICE_ACCOUNT_PATH =
-      "src/test/resources/integration_service_account.json";
+  /** Path to the admin service account credentials file. */
+  private static final String GOOGLE_SERVICE_ACCOUNT_ADMIN_PATH =
+      "src/test/resources/integration_service_account_admin.json";
 
-  public static ServiceAccountCredentials getGoogleCredentialsOrDie() {
+  /** Path to the admin service account credentials file. */
```
Comment (wchamber) [inline src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java:12]: fix comment. Maybe add 'The user service account doesn't have any permissions, but should be used as a test non-admin user to reference.'

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java [hunk 2]
```diff
@@ -5,16 +5,29 @@
 
 /** Provides cloud credentials to use in integration tests. */
 public class IntegrationCredentials {
-  /** Path to the service account credentials file. */
-  private static final String GOOGLE_SERVICE_ACCOUNT_PATH =
-      "src/test/resources/integration_service_account.json";
+  /** Path to the admin service account credentials file. */
```
Comment (wchamber) [inline src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java:8]: maybe add: 'The admin service account has the roles needed to operate the CRL APIs in the integration test project, e.g. create and delete resources.'

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java [hunk 3]
```diff
@@ -5,16 +5,29 @@
...
+
+  public static ServiceAccountCredentials getRegularUserGoogleCredentialsOrDie() {
     try {
-      return ServiceAccountCredentials.fromStream(new FileInputStream(GOOGLE_SERVICE_ACCOUNT_PATH));
+      return ServiceAccountCredentials.fromStream(new FileInputStream(GOOGLE_SERVICE_ACCOUNT_USER_PATH));
     } catch (Exception e) {
       throw new RuntimeException(
-          "Unable to load GoogleCredentials from " + GOOGLE_SERVICE_ACCOUNT_PATH + "\n", e);
+              "Unable to load GoogleCredentials from " + GOOGLE_SERVICE_ACCOUNT_USER_PATH + "\n", e);
     }
```
Comment (wchamber) [inline src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java]: optional: make a private helper `private ServiceAccountCredentials getGoogleCredentialsOrDice(String serviceAccountPath)` and use it for both admin and user.

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java [hunk 4]
```diff
@@ -5,16 +5,29 @@
...
+  public static ServiceAccountCredentials getAdminGoogleCredentialsOrDie() {
+    try {
+      return ServiceAccountCredentials.fromStream(new FileInputStream(GOOGLE_SERVICE_ACCOUNT_ADMIN_PATH));
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Unable to load GoogleCredentials from " + GOOGLE_SERVICE_ACCOUNT_ADMIN_PATH + "\n", e);
+    }
+  }
+
+  public static ServiceAccountCredentials getRegularUserGoogleCredentialsOrDie() {
```
Comment (wchamber) [inline src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java:26]: maybe omit 'Regular.' User is probably enough.

File: .github/workflows/build-and-test.yaml [hunk 5]
```diff
@@ -39,8 +41,11 @@ jobs:
             -e "VAULT_TOKEN=${{ steps.vault-token-step.outputs.vault-token }}" \
             -e "VAULT_ADDR=${VAULT_ADDR}" \
             vault:1.1.0 \
-            vault read -format json $VAULT_SERVICE_ACCOUNT_PATH \
-            | jq .data > $SERVICE_ACCOUNT_FILE
+            vault read -format json $VAULT_SERVICE_ACCOUNT_ADMIN_PATH \
+            | jq .data > $SERVICE_ACCOUNT_ADMIN_FILE &&#
+          docker run -it --rm -v $HOME:/root broadinstitute/dsde-toolbox:dev \
```
Comment (wchamber) [inline .github/workflows/build-and-test.yaml]: uh, should this match the docker run on lines 40-42?

File: .github/workflows/build-and-test.yaml [hunk 6]
```diff
@@ -32,15 +34,24 @@ jobs:
...
         run: |
           docker run --rm --cap-add IPC_LOCK \
             -e "VAULT_TOKEN=${{ steps.vault-token-step.outputs.vault-token }}" \
             -e "VAULT_ADDR=${VAULT_ADDR}" \
             vault:1.1.0 \
-            vault read -format json $VAULT_SERVICE_ACCOUNT_PATH \
-            | jq .data > $SERVICE_ACCOUNT_FILE
+            vault read -format json $VAULT_SERVICE_ACCOUNT_ADMIN_PATH \
+            | jq .data > $SERVICE_ACCOUNT_ADMIN_FILE
+      - name: Get service account credentials for regular user from Vault
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply .github/workflows/build-and-test.yaml]: done

File: README.md [hunk 7]
```diff
@@ -18,14 +18,22 @@ Build and run unit tests:
...
 docker run -it --rm -v $HOME:/root broadinstitute/dsde-toolbox:dev \
-  vault read -format json secret/dsde/terra/crl-test/default/service-account.json \
-  | jq .data > src/test/resources/integration_service_account.json
+  vault read -format json secret/dsde/terra/crl-test/default/service-account-admin.json \
+  | jq .data > src/test/resources/integration_service_account_admin.json \
+
+```
+and 
+```
+docker run -it --rm -v $HOME:/root broadinstitute/dsde-toolbox:dev
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply README.md]: done

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java (hunk 3 above)
Comment (yonghaoy) [reply src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java]: done. Die or Dice?

File: .github/workflows/build-and-test.yaml (hunk 5 above)
Comment (yonghaoy) [reply .github/workflows/build-and-test.yaml]: right, that is the out dated version. I used for local test

File: src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java (hunk 3 above)
Comment (wchamber) [reply src/test/java/bio/terra/cloudres/testing/IntegrationCredentials.java]: not "dice" :)

**Pull request merged**